patches/ungoogled-chromium/add-flag-to-show-avatar-button.patch

```diff
@@ -0,0 +1,96 @@
+     // Histograms" in tools/metrics/histograms/README.md (run the
+--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
++++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+@@ -206,12 +206,27 @@ void ToolbarView::Init() {
+       new BrowserActionsContainer(browser_, main_container, this);
+ 
+   if (ui::MaterialDesignController::IsRefreshUi()) {
++    const base::CommandLine& command_line = *base::CommandLine::ForCurrentProcess();
++    const std::string flag_value = command_line.GetSwitchValueASCII("show-avatar-button");
++
++    bool in_incognito_or_guest_mode = browser_->profile()->IsOffTheRecord() ||
++                                      browser_->profile()->IsGuestSession();
++
++    // Default behavior.
+     bool show_avatar_toolbar_button = true;
+ #if defined(OS_CHROMEOS)
+     // ChromeOS only badges Incognito and Guest icons in the browser window.
+-    show_avatar_toolbar_button = browser_->profile()->IsOffTheRecord() ||
+-                                 browser_->profile()->IsGuestSession();
++    show_avatar_toolbar_button = in_incognito_or_guest_mode;
+ #endif  // !defined(OS_CHROMEOS)
++
++    // Override according to show-avatar-button value.
++    if (flag_value == "always")
++      show_avatar_toolbar_button = true;
++    else if (flag_value == "incognito-and-guest")
++      show_avatar_toolbar_button = in_incognito_or_guest_mode;
++    else if (flag_value == "never")
++      show_avatar_toolbar_button = false;
++
+     if (show_avatar_toolbar_button)
+       avatar_ = new AvatarToolbarButton(browser_);
+   }
+--- a/chrome/browser/ui/views/frame/avatar_button_manager.cc
++++ b/chrome/browser/ui/views/frame/avatar_button_manager.cc
+@@ -31,6 +31,12 @@ void AvatarButtonManager::Update(AvatarB
+   BrowserFrame* frame = frame_view_->frame();
+   Profile* profile = browser_view->browser()->profile();
+ 
++  const base::CommandLine& command_line = *base::CommandLine::ForCurrentProcess();
++  const std::string flag_value = command_line.GetSwitchValueASCII("show-avatar-button");
++
++  bool in_incognito_or_guest_mode = profile->IsOffTheRecord() ||
++                                    profile->IsGuestSession();
++
+   // This should never be called in incognito mode.
+   DCHECK(browser_view->IsRegularOrGuestSession());
+   ProfileAttributesEntry* unused;
+@@ -42,7 +48,13 @@ void AvatarButtonManager::Update(AvatarB
+             ->GetProfileAttributesStorage()
+             .GetProfileAttributesWithPath(profile->GetPath(), &unused)) ||
+        // Desktop guest shows the avatar button.
+-       browser_view->IsIncognito())) {
++       browser_view->IsIncognito() ||
++       // Override according to show-avatar-button value.
++       (flag_value == "always") ||
++       (flag_value == "incognito-and-guest" && in_incognito_or_guest_mode)
++       ) &&
++      !(flag_value == "incognito-and-guest" && !in_incognito_or_guest_mode) &&
++      !(flag_value == "never")) {
```

Comment:
Why are the not conditions necessary (lines 92 and 93)? Isn't this covered by line 90?

Reply:
Redundant conditions are removed.